### PR TITLE
Bug 1003737 - Port and update /de/impressum/ to bedrock

### PR DESCRIPTION
**Pull request by malena**
https://bugzilla.mozilla.org/show_bug.cgi?id=1003737

### CODE REVIEW

File: bedrock/legal/templates/legal/impressum.html [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  <h1>Impressum</h1>
+</div>
+
+<div id="main-content">
+  <section class="intro">
+    <h2>Dienstanbieter:</h2>
+    <p>Mozilla Corporation (Gesellschaft mit beschränkter Haftung nach US-amerikanischem Recht) Corporate Number 2759208</p>
+
+    <h2>Sitz der Gesellschaften:</h2>
+    <p>313 E. Evelyn Ave., Mountain View CA 94041, USA</p>
```
Comment (schalkneethling) [inline bedrock/legal/templates/legal/impressum.html]: I feel that we should mark this up a bit more semantically. Something like here:
https://github.com/mozilla/bedrock/blob/master/bedrock/mozorg/templates/mozorg/contact/spaces/mountain-view.html#L11

File: media/css/legal/impressum.less [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+@import "../sandstone/lib.less";
+
+#tabzilla:before {
+    background-color: @textColorSecondary;
+}
+
+#main-feature {
+    > * {
```
Comment (schalkneethling) [inline media/css/legal/impressum.less]: This is a bit broad and, seeing that the only element in the main-feature container is a `h1`. In fact, I reckon you can just get rid of the `main-feature` container and move the `h1` into the main-content section.

File: media/css/legal/impressum.less [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    }
+}
+
+/* Tablet & Mobile */
+
+@media only screen and (max-width: @breakDesktop) {
+
+    #main-feature > *,
+    #main-content .intro,
+    #main-content .archive {
```
Comment (schalkneethling) [inline media/css/legal/impressum.less]: The .archive container does not seem to exist on the page anymore.

File: media/css/legal/impressum.less [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    }
+    .main-license {
+        .trailing-arrow();
+        .open-sans-light();
+        .font-size(18px);
+    }
+    .intro {
+        margin-bottom: @baseLine * 2;
+    }
+}
```
Comment (schalkneethling) [inline media/css/legal/impressum.less]: Looks like you could simplify this block to just:

```
#main-content {
    .intro {
        .span(8);
    }
}
```

File: media/css/legal/impressum.less [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+@media only screen and (max-width: @breakDesktop) {
+
+    #main-feature > *,
+    #main-content .intro,
+    #main-content .archive {
+        .span-all();
+    }
+
+}
```
Comment (schalkneethling) [inline media/css/legal/impressum.less]: Testing this in dev tools, it seems all we need for the CSS is the following:

```
#tabzilla:before {
    background-color: @textColorSecondary;
}
#main-content {
    h1 {
        padding: 20px 0 40px;
    }
    .intro {
        .span(8);
    }
}

@media only screen and (max-width: @breakDesktop) {
    #main-content {
        .intro {
            .span-all();
        }
    }
}
```
This of course assumes that ``#main-feature` is also removed from the HTML

Comment (malena) [inline media/css/legal/impressum.less]: Ahh thanks @schalkneethling! I took most of the styles from eula.less so glad to clean up.

**Comment by malena** [comment]
@schalkneethling I made some changes to the html. Hoping it's more semantic now. I need to remember that I have more flexibility to update semantics when porting over old php pages. Thanks!

File: media/css/legal/impressum.less [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    }
+}
+
+@media only screen and (max-width: @breakDesktop) {
+    #main-content {
+        .intro {
+            .span-all();
+        }
+    }
+}
```
Comment (schalkneethling) [inline media/css/legal/impressum.less]: Nit: new line at end of file.

File: bedrock/legal/templates/legal/impressum.html (hunk 1 above)
Comment (schalkneethling) [inline bedrock/legal/templates/legal/impressum.html]: Have you pushed your latest changes?

**Comment by malena** [comment]
@schalkneethling  Sorry it's committed now. Thanks!

File: bedrock/legal/templates/legal/impressum.html [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+{% block breadcrumbs %}{% endblock %}
+
+{% block extrahead %}
+  {% stylesheet 'legal-impressum' %}
+{% endblock %}
+
+{% block content %}
+
+
+<div id="main-content">
```
Comment (schalkneethling) [inline bedrock/legal/templates/legal/impressum.html]: Tiny Nit: remove two empty lines above the div?

**Comment by schalkneethling** [comment]
Looking good, thanks @malena r+wc :sunflower:

**Comment by sgarrity** [comment]
I wonder if we should add a redirect from `/de/impressum` -> `/de/legal/impressum` ?

**Comment by schalkneethling** [comment]
@sgarrity that makes sense yeah.

**Comment by sgarrity** [comment]
I think this should work?
`RewriteRule ^/de/impressum(/)?$ /de/legal/impressum/ [PT]`

**Comment by jgmize** [comment]
> I think this should work?
> RewriteRule ^/de/impressum(/)?$ /de/legal/impressum/ [PT]

If you're doing a redirect in apache instead of passing through to bedrock, you want the following:
`RewriteRule ^/de/impressum/?$ /de/legal/impressum/ [L,R=301]`

I also removed the parens around the optional trailing slash since you don't need to capture it.

**Comment by malena** [comment]
Thanks @sgarrity @jgmize. @schalkneethling updated

**Pull request merged**